controller/contour/deployment.go

```diff
@@ -0,0 +1,345 @@
+			Namespace: contour.Spec.Namespace.Name,
+			Name:      contour.Name,
+		},
+	}
+
+	if err := r.Client.Delete(ctx, deployment); err != nil {
+		if errors.IsNotFound(err) {
+			return nil
+		}
+		return err
+	}
+
+	r.Log.Info("deleted deployment", "namespace", deployment.Namespace, "name", deployment.Name)
+	return nil
+}
+
+// desiredDeployment returns the desired deployment for the provided contour using
+// image as Contour's container image.
+func desiredDeployment(contour *operatorv1alpha1.Contour, image string) (*appsv1.Deployment, error) {
+	parsedImage := strings.Split(image, ":")
+	labels := map[string]string{
+		"app.kubernetes.io/name":     "contour",
+		"app.kubernetes.io/instance": contour.Name,
+		// The image tag is used as the version.
+		"app.kubernetes.io/version":    parsedImage[1],
+		"app.kubernetes.io/component":  "ingress-controller",
+		"app.kubernetes.io/managed-by": "contour-operator",
+		// Associate the deployment with the provided contour.
+		owningContourLabel: contour.Name,
+	}
+
+	container := corev1.Container{
+		Name:            contourContainerName,
+		Image:           image,
+		ImagePullPolicy: corev1.PullIfNotPresent,
+		Command: []string{
+			"contour",
+		},
+		Args: []string{
+			"serve",
+			"--incluster",
+			"--xds-address=0.0.0.0",
+			"--xds-port=8001",
+			"--envoy-service-http-port=80",
+			"--envoy-service-https-port=443",
+			fmt.Sprintf("--contour-cafile=%s", filepath.Join("/", certsVolumeMountDir, "ca.crt")),
+			fmt.Sprintf("--contour-cert-file=%s", filepath.Join("/", certsVolumeMountDir, "tls.crt")),
+			fmt.Sprintf("--contour-key-file=%s", filepath.Join("/", certsVolumeMountDir, "tls.key")),
+			fmt.Sprintf("--config-path=%s", filepath.Join("/", cfgVolumeMountDir, cfgFileName)),
+		},
+		Env: []corev1.EnvVar{
+			{
+				Name: deployNsEnvVar,
+				ValueFrom: &corev1.EnvVarSource{
+					FieldRef: &corev1.ObjectFieldSelector{
+						APIVersion: "v1",
+						FieldPath:  "metadata.namespace",
+					},
+				},
+			},
+			{
+				Name: deployPodEnvVar,
+				ValueFrom: &corev1.EnvVarSource{
+					FieldRef: &corev1.ObjectFieldSelector{
+						APIVersion: "v1",
+						FieldPath:  "metadata.name",
+					},
+				},
+			},
+		},
+		Ports: []corev1.ContainerPort{
+			{
+				Name:          "xds",
+				ContainerPort: 8001,
+				Protocol:      "TCP",
+			},
+			{
+				Name:          "debug",
+				ContainerPort: 8000,
+				Protocol:      "TCP",
+			},
+		},
+		LivenessProbe: &corev1.Probe{
+			Handler: corev1.Handler{
+				HTTPGet: &corev1.HTTPGetAction{
+					Path: "/healthz",
+					Port: intstr.IntOrString{IntVal: int32(8000)},
+				},
+			},
+		},
+		ReadinessProbe: &corev1.Probe{
+			Handler: corev1.Handler{
+				TCPSocket: &corev1.TCPSocketAction{Port: intstr.IntOrString{IntVal: int32(8001)}},
+			},
+			InitialDelaySeconds: int32(15),
+			PeriodSeconds:       int32(10),
+		},
+		VolumeMounts: []corev1.VolumeMount{
+			{
+				Name:      certsVolumeName,
+				MountPath: certsVolumeMountDir,
```

Comment:
```suggestion
				MountPath: filepath.Join("/", certsVolumeMountDir),
```
Or include the leading slash in `certsVolumeMountDir` itself.